packages/api-plugin-orders/src/util/orderValidators/prepareOrder.js

```diff
@@ -0,0 +1,316 @@
+import SimpleSchema from "simpl-schema";
+import Random from "@reactioncommerce/random";
+import getAnonymousAccessToken from "@reactioncommerce/api-utils/getAnonymousAccessToken.js";
+import { Order as OrderSchema, orderInputSchema, paymentInputSchema } from "../../simpleSchemas.js";
+import validateInitialOrderData from "./validateInitialOrderData.js";
+import getDiscounts from "./getDiscounts.js";
+import getFinalFulfillmentGroups from "./getFinalFulfillmentGroups.js";
+import createPayments from "./createPayments.js";
+import getReferenceId from "./getReferenceId.js";
+import getCustomFields from "./getCustomFields.js";
+
+const inputSchema = new SimpleSchema({
+  "order": orderInputSchema,
+  "payments": {
+    type: Array,
+    optional: true
+  },
+  "payments.$": paymentInputSchema
+});
+
+/**
+ * @summary Formats validation error to common error format
+ * @param {Object} err Validation error object
+ * @returns {Object[]} Array of error entries
+ */
+function formatErrors(err) {
+  let errorEntries = [];
+  if (err.errorType === "ClientError" && err.error === "validation-error") {
+    errorEntries = (err.details || []).map((errorEntry) => {
+      const errorObject = {
+        errorName: "validation-error",
+        errorType: "Schema validation Error",
+        errorField: errorEntry.name || null,
+        fieldValue: errorEntry.value || null,
+        errorMessage: errorEntry.message || "No details"
+      };
+      return errorObject;
+    });
+  } else {
+    let errorField;
+    let fieldValue;
+    if (err.eventData) {
+      errorField = err.eventData.field;
+      fieldValue = err.eventData.value;
+    }
+    const errorObject = {
+      errorName: err.error,
+      errorType: "ReactionError",
+      errorField: errorField || null,
+      fieldValue: fieldValue || null,
+      errorMessage: err.reason || "No details"
+    };
+    errorEntries.push(errorObject);
+  }
+  return errorEntries;
+}
+
+/**
+ * @name prepareOrder
+ * @method
+ * @memberof Order/NoMeteorQueries
+ * @summary Validates if the input order details is valid and ready for order processing
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} input - order details, refer inputSchema
+ * @param {String} flag - flag which define if the call is from placeOrder or validateOrder
+ * @returns {Promise<Object>} output - order, token and validation results
+ */
+export default async function prepareOrder(context, input, flag) {
+  const cleanedInput = inputSchema.clean(input);
+  const validationResults = [];
+
+  // Step01: Input data validation against input schema
+  if (flag === "createOrderObject") {
+    inputSchema.validate(cleanedInput);
+  } else {
+    try {
+      inputSchema.validate(cleanedInput);
+    } catch (err) {
+      const validationErrors = formatErrors(err);
+      validationResults.push(...validationErrors);
+      return { errors: validationResults, success: false };
+    }
+  }
```

Comment:
This way looks a bit better. Same for `validateInitialOrderData` and other steps.

```suggestion
    try {
      inputSchema.validate(cleanedInput);
    } catch (err) {
      if (flag === "createOrderObject") {
        throw err;
      }    
      const validationErrors = formatErrors(err);
      validationResults.push(...validationErrors);
      return { errors: validationResults, success: false };
    }
  }
```